Adding docs for cover filter (#1493)
thumbor/thumbor: docs/cover.rst

Cover
========

Usage: `cover()`

Description
-----------

> This filter is used in gifs to extract the first frame of them as an image to be used as cover.

Reviewer: I suggest GIFs as a more accurate term (Wikipedia also uses it) and a small rewording:

```suggestion
This filter is used in GIFs to extract their first frame as the image to be used as cover.
```

Author: The suggestions were applied. Thank you!

> Obs: This filter will only function when `USE_GIFSICLE_ENGINE = True` are set to true on `thumbor.conf`

Reviewer: How about a Note as in other documents?

```suggestion
..note::
    This filter will only function when `USE_GIFSICLE_ENGINE = True` are set to true on `thumbor.conf`
```

Author: Changed to .. note::

Arguments
---------

No arguments.

Example
-------

.. image:: images/animated.gif
    :alt: Gif before cover filter

::

> .. code::

Reviewer: This line can be removed.

```suggestion
```

Author: Removed :)

> Cover
> ========
> 
> Usage: `cover()`
> 
> Description
> -----------
> 
> This filter is used in GIFs to extract their first frame as the image to be used as cover.
> 
> .. note::
> This filter will only function when `USE_GIFSICLE_ENGINE = True` are set to true on `thumbor.conf`
> 
> Arguments
> ---------
> 
> No arguments.
> 
> Example
> -------
> 
> .. image:: images/animated.gif
>     :alt: Gif before cover filter
> 
> ::
> 
>     `http://localhost:8888/unsafe/filters:cover()/http://server.my/animated_static.gif`
> 
> .. image:: images/animated_static.gif
>     :alt: Gif after cover filter

Reviewer: Sorry for missing this one.

👉 Tip: please, use the "Commit suggestion" button It's easier and the extra commit gets squashed once we're all happy with the changes 😊

```suggestion
Cover
=====

Usage: `cover()`

Description
-----------

This filter is used in GIFs to extract their first frame as the image to be used as cover.

.. note::

    This filter will only function when ``USE_GIFSICLE_ENGINE`` are set to ``True`` in ``thumbor.conf``:

.. code:: python

    USE_GIFSICLE_ENGINE = True

Arguments
---------

No arguments.

Example
-------

.. image:: images/animated.gif
    :alt: Gif before cover filter

::

    `http://localhost:8888/unsafe/filters:cover()/http://server.my/animated_static.gif`

.. image:: images/animated_static.gif
    :alt: Gif after cover filter
```
